hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java

```diff
@@ -111,6 +112,9 @@ public class HddsVolume extends StorageVolume {
   private AtomicInteger volumeTestFailureCount;
   private Queue<Boolean> volumeTestResultQueue;
 
+  private volatile CachedPendingDeletion cachedPendingDeletion = null;
```

Comment:
This data for all data volumes(HDDSVolume), so have it in each HddsVolume duplicate the function. 

Consider move it to ContainerController or ContainerSet.  

How much time will it take to iterate a 1 million ContainerSet?  If it takes too much time,  it's better to have a timerTask to refresh the data periodically and asynchronizely.